sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/SimpleReactiveCosmosRepository.java

```diff
@@ -28,22 +27,6 @@
     private final CosmosEntityInformation<T, K> entityInformation;
     private final ReactiveCosmosOperations cosmosOperations;
 
-    /**
-     * Initialization with metadata and applicationContext will create container if required
-     *
-     * @param metadata for entityInformation
-     * @param applicationContext for cosmosOperations
-     */
-    public SimpleReactiveCosmosRepository(CosmosEntityInformation<T, K> metadata,
-                                          ApplicationContext applicationContext) {
-        this.cosmosOperations = applicationContext.getBean(ReactiveCosmosOperations.class);
-        this.entityInformation = metadata;
-
-        if (this.entityInformation.isAutoCreateContainer()) {
```

Comment:
`AutoCreateContainer` is an annotation property - which supports creating containers automatically in cosmosdb if not exists. 
Why are we removing this functionality ?

Reply:
Because now we support multiple cosmos template, we update the `getTargetRepository` method in `ReactiveCosmosRepositoryFactory.java`, see [here](https://github.com/Azure/azure-sdk-for-java/pull/13197/files#diff-63ef672fd519ee2a15b8ebbb46ddea91R45), use `cosmosOperations` to replace `applicationContext`, we don't use applicationContext any more,  so I delete this constructor method

Reply:
Okay, I guess - we will have to find some other way to respect this annotation then, because otherwise containers will not be created when an entity will be initialized, no ?

Reply:
Now we use this constructor to create `SimpleReactiveCosmosRepository`,
```java
public SimpleReactiveCosmosRepository(CosmosEntityInformation<T, K> metadata,
                                          ReactiveCosmosOperations reactiveCosmosOperations) {
        this.cosmosOperations = reactiveCosmosOperations;
        this.entityInformation = metadata;

        if (this.entityInformation.isAutoCreateContainer()) {
            createContainerIfNotExists();
        }
    }
```
customers can still use annotation property to decide to create containers or not

Reply:
Sounds good, thank you for explaining this. 

By the way - here is the PR that brings azure-spring-data-cosmos to azure-cosmos dependency V4 - which is track 2 library : https://github.com/Azure/azure-sdk-for-java/pull/13229